### PR TITLE
feat: Allow setting reverse filter p threshold

### PR DESCRIPTION
**Pull request by paulgessinger**
We previously didn't set up the reverse filtering at all in the examples track fitting algorithm, so the WARNING in #1101 and #1077 was never observed.

This PR exposes a momentum threshold variable through the config struct and into python. The pytests run it with a threshold of 0 (no reverse filtering at all) and 1 TeV (almost always reverse filtering), so we should at least see this warning here until #1101 is merged.

### CODE REVIEW

File: Examples/Scripts/Python/truth_tracking.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def runTruthTracking(
     outputDir: Path,
     digiConfigFile: Path,
     directNavigation=False,
+    reverseFilteringMomThreshold=0*u.GeV,
```
Comment (asalzburger) [inline Examples/Scripts/Python/truth_tracking.py:17]: This effectively switches reverse filtering off, right?

**Review by asalzburger — APPROVED** [review]
Makes totally sense.

**Comment by asalzburger** [comment]
There's still a failure & the format checks are running.

File: Examples/Scripts/Python/truth_tracking.py (hunk 1 above)
Comment (paulgessinger) [reply Examples/Scripts/Python/truth_tracking.py:17]: Exactly, which is what the previous default was.

**Comment by paulgessinger** [comment]
> There's still a failure & the format checks are running.

The failure is because #1101 is not in this branch yet. I'll update and it should resolve the failure (as intended).

**Comment by paulgessinger** [comment]
I also fixed the format now.

**Comment by paulgessinger** [comment]
Ok, everything seems to succeed.
@asalzburger can you approve again?

**Comment by paulgessinger** [comment]
The CI failure was likely caused by a regression introduced in #1103. This should be fixed in #1115. This branch now includes that commit. The CI should run through now (I hope), but I'll mark this one WIP again until #1115 is merged.

**Comment by paulgessinger** [comment]
Tests succeeds now it seems. @asalzburger can you approve one more time?

**Comment by paulgessinger** [comment]
@asalzburger The bot isn't merging because the approval itself is not valid yet.

**Comment by asalzburger** [comment]
Merged by hand.

**Pull request merged**